src/methods/summarize.ts

```diff
@@ -1,18 +1,18 @@
 import log from "../helpers/log.js"
-import { SimpleDataItem } from "../types/SimpleData.types.js"
+import { SimpleDataItem, SimpleDataValue } from "../types/SimpleData.types.js"
 import { flatRollup, mean, sum, median, max, min, deviation } from "d3-array"
 import checkTypeOfKey from "../helpers/checkTypeOfKey.js"
 import isEqual from "lodash.isequal"
 import hasKey from "../helpers/hasKey.js"
 
 export default function summarize(
-    data: SimpleDataItem[], 
-    value: any, 
-    keyCategories: any, 
-    summary: any, 
-    weight: any, 
-    verbose: boolean, 
-    nbValuesTested: number, 
+    data: SimpleDataItem[],
+    keyValue: string | string[],
+    keyCategories: string | string[] | undefined,
+    summary: string | string[] | undefined,
+    weight: string | undefined,
```

Comment:
I would rather use optional arguments instead of undefined so we do not have to define them if's not needed
```suggestion
    keyCategories?: string | string[],
    summary?: string | string[],
    weight?: string,
```